File: app/src/main/java/online/tatarintsev/currencyrates/presenter/CurrencyPresenter.java
package online.tatarintsev.currencyrates.presenter;

import android.os.Bundle;
import android.view.View;

public interface CurrencyPresenter {
    void onCreate(Bundle savedInstanceState);
    void onSaveInstanceState(Bundle outState);

    void onAttachView(CurrencyView view);
    void onDetachView();

    void onUserAction(View view);
}

File: app/src/main/java/online/tatarintsev/currencyrates/presenter/CurrencyPresenter.kt
package online.tatarintsev.currencyrates.presenter

import android.os.Bundle
import android.view.View

interface CurrencyPresenter {
    fun onCreate(savedInstanceState: Bundle?)
    fun onSaveInstanceState(outState: Bundle)

    fun onAttachView(view: CurrencyView)
    fun onDetachView()

    fun onUserAction(view: View)
}